tests/test_learning.py

```diff
@@ -12,3 +12,31 @@ def test_weighted_mode():
 
 def test_weighted_replicate():
     assert weighted_replicate('ABC', [1, 2, 1], 4) == ['A', 'B', 'B', 'C']
+
```

Comment:
I really appreciate your work. This is just a small mistake.
You are required to import `rms_error()`, `manhattan_distance()`, `mean_boolean_error()` and `mean_error()` from learning.